Assets/Scripts/Dialogue/DialogueSystem.cs

```diff
@@ -29,10 +44,29 @@ private void Update()
         if (CheckIfAdvanceKeyPressed())
             textbox.FastForward();
     }
+    // Helper function
+    public void enableSkip(bool enabled) {
+        skipEnabled = enabled;
+    }
 
     private bool CheckIfAdvanceKeyPressed()
     {
-        return Input.GetKeyDown(advanceKey) || Input.GetMouseButtonDown(0);
+        if(Input.GetKeyDown(advanceKey) || Input.GetMouseButtonDown(0)) {
+            // When the player attempts to advance, create this pointer event thing.
+            // Raycast from the point clicked at, and check UI objects hit.
+            // If any object hit is the skip button, then return false and don't skip.
+            // Otherwise, return if we can advance or not.
+            pointerData = new PointerEventData(eventSys);
+            pointerData.position = Input.mousePosition;
+            List<RaycastResult> results = new List<RaycastResult>();
+            raycaster.Raycast(pointerData, results);
+            foreach (RaycastResult result in results)
+            {
+                if(result.gameObject.tag == "SkipButton") return false;
+            }
+            return skipEnabled;
+        }
+        return false;
     }
```

Comment:
this works for the most part, but there is a bug where hovering over "skip" and pressing "space" won't advance the dialog. also, i'm concerned that this method isn't very scalable. if we had multiple buttons, we'd have to do all this for each one-- not the best practice.

instead of casting a ray ourselves and seeing if it hits, let's rely on the built-in raycasting system from unity UI. then we can remove Input.GetMouseButtonDown() so we don't check the click globally. instead, we can use an invisible full-screen button used as a trigger/condition for advancing the dialog. it can be layered between the textbox and the skip button so it won't capture the click when the skip button (or anything above it) is clicked.

this change shouldn't require any modifications to the scene, just the dialogue system script and prefab.